st2common/tests/unit/test_action_alias_utils.py

```diff
@@ -14,293 +14,64 @@
-        self.assertEqual(value, '{"c": "d"}')
-
-        start = len(stream) - 2
-        self.assertFalse(JsonValueParser.is_applicable(stream[start]), 'Should not be parsable.')
-
-    def testEndJsonParsing(self):
-        stream = 'some meaningful {"a": "b"}'
-
-        start = len('some meaningful ')
-        self.assertTrue(JsonValueParser.is_applicable(stream[start]), 'Should be parsable.')
-        _, value, _ = JsonValueParser.parse(start, stream)
-        self.assertEqual(value, '{"a": "b"}')
-
-    def testComplexJsonParsing(self):
-        stream = 'some meaningful {"a": "b", "c": "d", "e": {"f": "g"}, "h": [1, 2]}'
-
-        start = len('some meaningful ')
-        self.assertTrue(JsonValueParser.is_applicable(stream[start]), 'Should be parsable.')
-        _, value, _ = JsonValueParser.parse(start, stream)
-        self.assertEqual(value, '{"a": "b", "c": "d", "e": {"f": "g"}, "h": [1, 2]}')
-
-    def testIncompleteStringParsing(self):
-        stream = 'some meaningful {"a":'
-
-        start = len('some meaningful ')
-        self.assertTrue(JsonValueParser.is_applicable(stream[start]), 'Should be parsable.')
-        try:
-            JsonValueParser.parse(start, stream)
-            self.assertTrue(False, 'Parsing failure expected.')
-        except content.ParseException:
-            self.assertTrue(True)
+from st2common.models.utils.action_alias_utils import ActionAliasFormatParser
 
 
 class TestActionAliasParser(TestCase):
-    def test_default_key_value_param_parsing(self):
-        # Empty string
+    def testEmptyString(self):
         alias_format = ''
         param_stream = ''
         parser = ActionAliasFormatParser(alias_format, param_stream)
         extracted_values = parser.get_extracted_param_value()
         self.assertEqual(extracted_values, {})
 
-        # 1 key value pair provided in the param stream
+    def testArbitraryPairs(self):
+        # single-word param
         alias_format = ''
         param_stream = 'a=foobar1'
         parser = ActionAliasFormatParser(alias_format, param_stream)
         extracted_values = parser.get_extracted_param_value()
         self.assertEqual(extracted_values, {'a': 'foobar1'})
 
+        # multi-word double-quoted param
         alias_format = ''
-        param_stream = 'foo a=foobar2 poonies bar'
-        parser = ActionAliasFormatParser(alias_format, param_stream)
-        extracted_values = parser.get_extracted_param_value()
-        self.assertEqual(extracted_values, {'a': 'foobar2'})
-
-        # Multiple params provided
-        alias_format = ''
-        param_stream = 'a=foobar1 b=boobar2 c=coobar3'
-        parser = ActionAliasFormatParser(alias_format, param_stream)
-        extracted_values = parser.get_extracted_param_value()
-        self.assertEqual(extracted_values, {'a': 'foobar1', 'b': 'boobar2', 'c': 'coobar3'})
-
-        # Multiple params provided
-        alias_format = ''
-        param_stream = 'a=foobar4 b=boobar5 c=coobar6'
```

Comment:
As agreed, we do not support arbitrary pairs mixed with normal strings, i.e. `mixed a=foobar1 some more b=boobar2 text c=coobar3 yeah`. We discussed this with @jfryman and @manasdk, and agreed this is a rare case which we don't really need. Continuing to support them would result in unexpected (and, honestly, illogical) behavior in edge cases.

Rest of the tests are still there, and I've added a couple new ones, too.